js/src/forum/components/DiscussionPage.js

```diff
@@ -95,30 +81,32 @@ export default class DiscussionPage extends Page {
     );
   }
 
-  onbeforeupdate(vnode) {
-    super.onbeforeupdate(vnode);
-
-    if (m.route.get() !== this.prevRoute) {
-      this.onNewRoute();
-      this.prevRoute = m.route.get();
+  onNewRoute() {
+    // If we have routed to the same discussion as we were viewing previously,
+    // prompt the post stream to jump to the new 'near' param.
+    // Otherwise, load in a new discussion. The `else` branch will
+    // be followed when `onNewRoute` is called from `oninit`.
+    const idParam = m.route.param('id');
+    if (this.discussion && idParam && idParam.split('-')[0] === this.discussion.id()) {
```

Comment:
I'd split this out in this PR already, since the diff is very hard to read. Something like this:

```js 

isSameRoute() {
    const idParam = m.route.param('id');
    return this.discussion && idParam && idParam.split('-')[0] === this.discussion.id();
}

onSameRoute() {
    const near = m.route.param('near') || '1';

    if (near !== String(this.near)) {
      this.stream.goToNumber(near);
    }

    this.near = near;
}

```

and:
```js

onNewRoute() {
   if (this.isSameRoute()) return this.onSameRoute();
   
    super.onNewRoute();

    ....

}

```